packages/effect/src/internal/resource.ts

```diff
@@ -46,11 +47,18 @@ export const manual = <A, E, R>(
   core.flatMap(core.context<R>(), (env) =>
     pipe(
       scopedRef.fromAcquire(core.exit(acquire)),
-      core.map((ref) => ({
-        [ResourceTypeId]: resourceVariance,
-        scopedRef: ref,
-        acquire: core.provideContext(acquire, env)
-      }))
+      core.map((ref) => {
+        const resource = {
```

Comment:
I think we can move the prototype out of here and use `Object.create`.

Reply:
sure